### ISSUE TITLE
VSCode Multiple cursors

### ISSUE THREAD

**Issue by mizagog** [asvetliakov/vscode-neovim#97]
Not a bug realy, but is there plans to do multiple cursours with VS's cursours? it is extremely useful to create cursors for occurrences and then be able to do vim stuff for each cursor.

**Comment by avstudio** [comment]
Yeah, I'm looking the way how to select in visual, and then with Ctrl+n add to selection other occurrences like in vim-multiple-cursors. Is it feasible right now? Btw. great work.

**Comment by asvetliakov** [comment]
not yet, but definitely something i will look into. For now they're work in insert mode. In meantime```cgn``` (```.``` dot repeat from ```vim-repeat``` doesn't work though yet) or ```dgn``` may work for you

**Comment by mizagog** [comment]
I will check it out

**Comment by pumpkinlink** [comment]
What I did was to remove the Ctrl+d keybinding for "outdent line", now I can use VSCode's native keybinding for "add selection to next find match" (only on insert mode of course)

**Comment by brentluna** [comment]
> What I did was to remove the Ctrl+d keybinding for "outdent line", now I can use VSCode's native keybinding for "add selection to next find match" (only on insert mode of course)
@pumpkinlink where did you make this change? I saw the setting is in a package.json in the extension code, but don't know how i would access that.

**Comment by echaya** [comment]
wish we could support 'mg979/vim-visual-multi' natively but may need some 'vscode-easymotion' fork

**Comment by yousefamar** [comment]
> In meantime `cgn` or `dgn` may work for you

@asvetliakov could you explain what you mean by this? Thanks!

**Comment by theol0403** [comment]
> > In meantime `cgn` or `dgn` may work for you
> 
> @asvetliakov could you explain what you mean by this? Thanks!

http://vimcasts.org/episodes/operating-on-search-matches-using-gn/

**Comment by pencilcheck** [comment]
> looking forward to the enhancement. For anyone interested in using `vim-visual-multi` as a temp solution, please check out: [#409 (comment)](https://github.com/vscode-neovim/vscode-neovim/issues/409#issuecomment-830796423)

Thanks! still buggy when trying to edit selections but at least C-n seems to select multiple regions now

**Comment by AndroAmater** [comment]
Using vim-visual-multi and editing the selection doesn't give live feedback and in some cases doesn't work at all for me. 

It would be great if we could have this! It's probably the only vscode feature that I'm missing and I miss it quite a lot tbh. 

Multi cursors allow for super powerful transformations with very little overhead. The built in version is very welcome, but it falls short for me. If we could at least have the ctrl+d functionality from vs code it would be great, but I'm guessing it's easier said than done since nvim doesn't really handle text in the same way as vscode. Fixing this plugin would be more than good enough for me!

Hope support for this plugin or vscode multicursors will be implemented. Tnx for the great job with the plugin so far. 

Really loving it :smiley:!

**Comment by Spirarel** [comment]
Also would love to see this hit parity with the terminal experience!

In the meantime, is it possible to remap the existing multiple-cursor commands?
—I use 'm' pretty extensively already.

**Comment by 55Cancri** [comment]
please let us know @asvetliakov any challenges or blockers you have with implementing this feature. Also willing to fund the creation of this feature if that is the bottleneck here with making otherwise great extension perfect

**Comment by xiyaowong** [comment]
After the next release, plugin [vscode-neovim/vscode-multi-cursor.nvim](https://github.com/vscode-neovim/vscode-multi-cursor.nvim) will be ready for use. Not sure if it meets everyone’s needs, but it definitely makes multi-cursor easier to use.
